src/classes/modules/java.base/java/lang/Class.java

```diff
@@ -125,6 +125,10 @@ public InputStream getResourceAsStream (String name) {
 
   public URL getResource (String rname) {
     String resolvedName = getResolvedName(rname);
+
+    if (resolvedName.startsWith("java/") || resolvedName.startsWith("jdk/") || resolvedName.startsWith("sun/"))
```

Comment:
Let's use ClassLoader.getModule to determine what prefix should be used, and use that as an extra parameter to getResource. This is how it seems to work in JDK 11, and avoids introducing a different behavior in JPF.

Reply:
If the module is unnamed (getModule().isNamed() returns false), then we just load the resource as below, without a module. Otherwise, we use the module of the given class loader as the module name (the additional parameter).

Reply:
Module.isName is missing in model class of Module class. Hence, as a first step should I add and implement the missing method in the model class.

Reply:
Yes, please do that. Perhaps you can get the right module name from the host JVM.